src/lib/internal.ts

```diff
@@ -19,12 +22,12 @@ export function generateActionObject<T>(
   store: Type<EntityState<T>>,
   payload?: any
 ) {
-  const name = store['NGXS_META'].path;
+  const name = store[NGXS_META_KEY].path;
   const ReflectedAction = function(data: T) {
     this.payload = data;
   };
   const obj = new ReflectedAction(payload);
-  obj.__proto__.constructor.type = `[${name}] ${fn}`;
+  Reflect.getPrototypeOf(obj).constructor['type'] = `[${name}] ${fn}`;
```

Comment:
This seems like not secure code

Reply:
Do you have a proposal? This is the one line that makes the whole library work.

Reply:
Since type seems to not be a key on constructor, the incoming change looks better in my opinion.

Reply:
I prefer the incoming change too. I don't think it is less secure.

Reply:
@odahcam I have just noticed that your username is your surname backwards! 🤯

Reply:
And you don't even speak portuguese, congratulations, maybe you are the most clever person to find this out yet. 😆